planner/core/find_best_task.go

```diff
@@ -330,21 +330,21 @@ func (op *physicalOptimizeOp) appendChildToCandidate(candidateInfo *tracing.Phys
 }
 
 func (op *physicalOptimizeOp) setBest(lp LogicalPlan, pp PhysicalPlan, prop string) {
-	if op == nil || op.tracer == nil {
+	if op == nil || op.tracer == nil || pp == nil {
 		return
 	}
 	traceInfo := op.tracer.State[fmt.Sprintf("%v_%v", lp.TP(), lp.ID())][prop]
 	if traceInfo == nil {
 		return
 	}
-	traceInfo.BestTask = &tracing.PhysicalPlanTrace{ID: pp.ID(), TP: pp.TP(), Info: pp.ExplainInfo(), Cost: pp.Cost()}
+	traceInfo.BestTask = &tracing.PhysicalPlanTrace{ID: pp.ID(), TP: pp.TP(), Cost: pp.Cost(), Info: pp.ExplainInfo()}
 }
 
 func (op *physicalOptimizeOp) appendCandidate(logicalPlan *baseLogicalPlan, physicalPlan PhysicalPlan, prop string) *tracing.PhysicalPlanTrace {
 	if op == nil || op.tracer == nil {
 		return nil
 	}
-	PhysicalPlanTrace := &tracing.PhysicalPlanTrace{TP: physicalPlan.TP(), ID: physicalPlan.ID(), Info: physicalPlan.ExplainInfo()}
```

Comment:
during `appendCandidate`, The info might be not initialized that would cause panic here